docs/src/adr/019adr-changelog-management.md

```diff
@@ -0,0 +1,182 @@
+    commits or designate special PR labels to work around this interference?
+
+
+### 2. Use Unclog
+
+@thanethompson has written the [unclog][] utility, which is used in both
+[informalsystems/tendermint-rs][] and [informalsystems/ibc-rs][]. The tool
+stores changelog entries in a directory structure, with each entry in its own
+file, ensuring merge conflicts can't arise. 
+
+Entries are added via the CLI. E.g.:
+
+```sh
+unclog add --id some-new-feature \
+  --issue 23 \
+  --section breaking-changes \
+  --message "Some *new* feature"
+```
+
+If the `--messages` argument is omitted, it will open your configured editor for
+authoring the message.
+
+The tool has subcommands to generate and update the `CHANGES.md`, and it supports
+a variety of options via its TOML configuration.
+
+[unclog]: https://github.com/informalsystems/unclog
+[informalsystems/tendermint-rs]: https://github.com/informalsystems/tendermint-rs
+[informalsystems/ibc-rs]: https://github.com/informalsystems/ibc-rs
+
+#### Pros
+
+- Requires no new development
+- It is developed internally to informal
+
+#### Cons
+
+- Adds cognitive and procedural overhead to adding changelog entries
+- Would add a rust dependency to our devenv
+- Requires external contributors to either install a CLI tool or figure out a
+  relatively complex file and folder structure to add changelog entries
+
+### 3. Use a custom git merge driver
+
+If we wrote a custom merge driver to work on the simple changelog format we
+could continue our current practice, and just fall back to the custom merge
+driver in case of merge conflicts.
+
+#### Pros
+
+- Allows us to keep our current process of simply updating a markdown file,
+  with which, moreover, most devs are already familiar
+
+#### Cons
+
+- Could require external dependencies to be installed, depending on
+  implementation.
+- It's not possible to configure custom merge drivers for github, so we'd either
+  need to develop a github action or bot to monitor for merge conflicts in the
+  changelog and merge them automatically, or devs would be back to having to
+  resolve merge conflicts locally.
+
+### 4. Write some lightweight tooling
+
+We could author some simple tooling that would capture the core behavior of
+`unclog`, using files in directories, but neither require external dependencies
+nor knowledge of a new CLI.
+
+#### Pros
+
+- No new tooling to learn
+- No external dependencies to integrate
+
+#### Cons
+
+- Would require some development time (=< 1 day)
+- Would require contributors to understand the folder structure
+
+## Solution
+
+We have opted for option 4. Despite the added cost of development, we think the
+simplicity of use and maintenance will offset the development cost.
+
+### Design
+
+We will introduce a new directory tree to the project:
+
+```sh
+.unreleased/
+├── breaking-changes/
+├── bug-fixes/
+├── documentation/
+└── features/
+```
+
+Each change will be added as a single markdown file in the appropriate
+directory.
```

Comment:
What is a "change" in this context? A markdown bullet list?

Reply:
It is sometimes beneficial to keep some logical order among changelog entries, it would be good to support this in the future.

Reply:
I was thinking an entry describing a change would just be a paragraph or two of markdown. Different entries should be different files. No point in having bullet points inside the files afaict, that seems confusing to me.

Good point about order. I think we can just assemble the log entries in lexigraphic order of the file names. So that can be used to control the ordering within a section.

Reply:
> I was thinking an entry describing a change would just be a paragraph or two of markdown. [...] No point in having bullet points inside the files afaict, that seems confusing to me.

I'm just wondering how this will render in the combined changelog file, but we can leave this for implementation.